src/commands/treeSitterMapping.ts

```diff
@@ -13,77 +24,321 @@ const nodeTypeFiles = new Map([
-                        if (
-                            !expressionMappings.get(subNodeType.type)?.languages.includes(language)
-                        ) {
-                            expressionMappings.get(subNodeType.type)?.languages.push(language);
+    for (const nodeType of nodeTypes) {
+        // Ignore all unnamed syntax nodes that are no binary expressions (as we require them) to get a "kind of"
+        // abstract syntax tree by removing syntax details.
+        // See also https://tree-sitter.github.io/tree-sitter/using-parsers#named-vs-anonymous-nodes
+        if (!nodeType.named) {
+            dlog("Excluded unnamed node type " + nodeType.type + " for language " + languageAbbr);
+            continue;
+        }
+
+        if (nodeType.type === "binary_expression") {
+            if (nodeType?.fields?.operator?.types !== undefined) {
+                for (const binaryOperatorType of nodeType.fields.operator.types) {
+                    const { type: binaryOperator } = binaryOperatorType;
+                    const mapKey = nodeType.type + "_" + binaryOperator;
+                    removedNodeTypes.delete(mapKey);
+
+                    const expression = expressionMappings.get(mapKey);
+                    if (expression !== undefined) {
+                        if (!expression.languages.includes(languageAbbr)) {
+                            changelog.addedNodeToLanguage(expression, languageAbbr);
+                            expression.languages.push(languageAbbr);
+                            dlog(
+                                "Language " +
+                                    languageAbbr +
+                                    ' was added to node type "' +
+                                    mapKey +
+                                    '".'
+                            );
                         }
                     } else {
-                        expressionMappings.set(subNodeType.type, {
-                            expression: subNodeType.type,
+                        expressionMappings.set(mapKey, {
+                            expression: mapKey,
                             metrics: [],
                             type: "statement",
-                            category: "",
-                            languages: [language],
+                            category: "binary_expression",
+                            languages: [languageAbbr],
+                            operator: binaryOperator,
                         });
+                        changelog.addedNewNode(mapKey, languageAbbr);
+                        dlog(
+                            'New node type "' +
+                                mapKey +
+                                '" was added for language "' +
+                                languageAbbr +
+                                '".'
+                        );
+                    }
+                }
+            }
+            continue;
+        }
```

Comment:
Perfect candiate to be extracted into a separate method.

Reply:
Splitted updateLanguage into separate methods